### PR TITLE
[image][android] Add `resizeMode: "repeat"` support

### PR DESCRIPTION
**Pull request by barthap**
# Why

To mirror RN `<Image resizeMode="repeat" />` support.

# How

Glide doesn't support repeatable image out of the box. 
What it does is to provide image "resource" into "image view" (`ExpoImageView`).
In our case, the resource is always `android.graphics.Drawable`, because it works with all supported file formats. Other options could be `File`, `Gif` or `Bitmap`.
The drawable is injected into view during `.into(this)` call, but not directly. A [`DrawableImageViewTarget` instance is created](https://github.com/bumptech/glide/blob/master/library/src/main/java/com/bumptech/glide/request/target/ImageViewTargetFactory.java), which is a Glide "target" for `ImageView` components. It [internally uses `view.setImageDrawable(drawable)`](https://github.com/bumptech/glide/blob/10acc31a16b4c1b5684f69e8de3117371dfa77a8/library/src/main/java/com/bumptech/glide/request/target/DrawableImageViewTarget.java#L24) to finally update the view.
There are two places where we can disrupt this process:
1. Create a custom target: `ImaveViewTarget<Drawable>` implementation and manually call `view.setImageDrawable()` there.
2. Override `ImageView.setImageDrawable()` inside `ExpoImageView` as it is called anyway in Glide's internal `DrawableImageViewTarget` implementation. I like this approach more, because it avoids anonymous objects.

`Drawable` is abstract base class for both `BitmapDrawable` and `PictureDrawable` implementations. Only the former supports image tiling (repeating), but some formats (including SVG - [see `SVGDrawableTranscoder.kt`](https://github.com/expo/expo/blob/9b7b1ecefd6bc9d6424acd17d896f7e0712a1d3f/packages/expo-image/android/src/main/java/expo/modules/image/svg/SVGDrawableTranscoder.kt#L20)) give us instance of the latter. So a conversion Picture->Bitmap drawable is needed.
Unfortunately, this conversion requires creating a canvas and re-drawing the image, which may have negative performance impact, but not tested yet

# Test Plan

- [x] bare-expo Image NCL: 
Screenshot 2021-08-31 at 12 02 10

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).

### CODE REVIEW

File: packages/expo-image/android/src/main/java/expo/modules/image/ExpoImageView.kt [hunk 1]
```diff
@@ -139,6 +143,7 @@ class ExpoImageView(
       progressInterceptor.registerProgressListener(sourceToLoad.toStringUrl(), eventsManager)
       eventsManager.onLoadStarted()
       requestManager
+        .asDrawable()
```
Comment (barthap) [inline packages/expo-image/android/src/main/java/expo/modules/image/ExpoImageView.kt:146]: Calling `requestManager.load()` is a short form of `requestManager.asDrawable().load()` - I wanted to explicitly specify we want a `Drawable`.

Other options would be `asFile()`, `asGif()` or `asBitmap()` - unfortunately the last one is probably not supported by all file formats

File: packages/expo-image/android/src/main/java/expo/modules/image/ExpoImageView.kt [hunk 2]
```diff
@@ -234,5 +235,18 @@ class ExpoImageView(
       }
     }
   }
+
+  /**
+   * Called when Glide "injects" drawable into the view
+   */
+  override fun setImageDrawable(drawable: Drawable?) {
```
Comment (bbarthec) [inline packages/expo-image/android/src/main/java/expo/modules/image/ExpoImageView.kt:242]: Ass a comment here to explain why we're doing this 👀 
I know that it's pretty obvious after reading the method body, but let's have it in plain english here as well 🤞

**Review by bbarthec — APPROVED** [review]
It looks very solid from my point of view 😉 Great job!

**Review by mstach60161 — APPROVED** [review]
Looks good! 👌
